spinalcordtoolbox/moco.py

```diff
@@ -0,0 +1,894 @@
+    else:
+        metric_radius = '4'
+    file_out_concat = file_out
+
+    kw = dict()
+    im_data = Image(file_src)  # TODO: pass argument to use antsReg instead of opening Image each time
+
+    # register file_src to file_dest
+    if param.todo == 'estimate' or param.todo == 'estimate_and_apply':
+        # If orientation is sagittal, use antsRegistration in 2D mode
+        # Note: the parameter --restrict-deformation is irrelevant with affine transfo
+        if im_data.orientation[2] in 'LR':
+            cmd = ['isct_antsRegistration',
+                   '-d', '2',
+                   '--transform', 'Affine[%s]' %param.gradStep,
+                   '--metric', param.metric + '[' + file_dest + ',' + file_src + ',1,' + metric_radius + ',Regular,' + param.sampling + ']',
+                   '--convergence', param.iter,
+                   '--shrink-factors', '1',
+                   '--smoothing-sigmas', param.smooth,
+                   '--verbose', '1',
+                   '--output', '[' + file_mat + ',' + file_out_concat + ']']
+            cmd += sct.get_interpolation('isct_antsRegistration', param.interp)
+            if im_mask is not None:
+                # if user specified a mask, make sure there are non-null voxels in the image before running the registration
+                if np.count_nonzero(im_mask.data):
+                    cmd += ['--masks', im_mask.absolutepath]
+                else:
+                    # Mask only contains zeros. Copying the image instead of estimating registration.
+                    sct.copy(file_src, file_out_concat, verbose=0)
+                    do_registration = False
+                    # TODO: create affine mat file with identity, in case used by -g 2
+        # 3D mode
+        else:
+            cmd = ['isct_antsSliceRegularizedRegistration',
+                   '--polydegree', param.poly,
+                   '--transform', 'Translation[%s]' %param.gradStep,
+                   '--metric', param.metric + '[' + file_dest + ',' + file_src + ',1,' + metric_radius + ',Regular,' + param.sampling + ']',
+                   '--iterations', param.iter,
+                   '--shrinkFactors', '1',
+                   '--smoothingSigmas', param.smooth,
+                   '--verbose', '1',
+                   '--output', '[' + file_mat + ',' + file_out_concat + ']']
+            cmd += sct.get_interpolation('isct_antsSliceRegularizedRegistration', param.interp)
+            if im_mask is not None:
+                cmd += ['--mask', im_mask.absolutepath]
+        # run command
+        if do_registration:
+            kw.update(dict(is_sct_binary=True))
+            env = dict()
+            env.update(os.environ)
+            env = kw.get("env", env)
+            # reducing the number of CPU used for moco (see issue #201)
+            env["ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS"] = "1"
+            status, output = sct.run(cmd, verbose=1 if param.verbose == 2 else 0, **kw)
```

Comment:
it looks like `status`, `output` are never checked?

Reply:
indeed-- in the past we used the output, but it was not reliable as mentioned in my comment above: https://github.com/neuropoly/spinalcordtoolbox/pull/2634#discussion_r402536239.

for legacy reasons the output is still there even though not used. 

For future code review i've added an explanation about the file checking approach (vs. status): 0df467d54debe6d1634917e8cc48e920e3d1c77b